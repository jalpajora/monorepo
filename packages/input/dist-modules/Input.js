'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports['default'] = void 0;

var _react = _interopRequireDefault(require('react'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {default: obj};
}

var Input = function Input(_ref) {
  var _ref$value = _ref.value,
    value = _ref$value === void 0 ? '' : _ref$value,
    _ref$type = _ref.type,
    type = _ref$type === void 0 ? 'text' : _ref$type,
    placeholder = _ref.placeholder,
    _ref$onChange = _ref.onChange,
    onChange =
      _ref$onChange === void 0
        ? function () {
            return console.log('changed');
          }
        : _ref$onChange;
  return /*#__PURE__*/ _react['default'].createElement('input', {
    onChange: onChange,
    placeholder: placeholder,
    type: type,
    value: value,
  });
};

var _default = Input;
exports['default'] = _default;
