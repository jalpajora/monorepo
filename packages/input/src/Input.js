import React from 'react';

export default function Input({value = '', type = 'text', placeholder, onChange = () => console.log('changed')}) {
  return <input type={type} placeholder={placeholder} value={value} onChange={onChange} />;
}
