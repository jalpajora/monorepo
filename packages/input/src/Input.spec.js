import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

describe('Input', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Input placeholder="user@gmail.com" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});