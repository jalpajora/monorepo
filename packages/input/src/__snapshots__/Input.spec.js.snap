// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="user@gmail.com"
  type="text"
  value=""
/>
`;
